commons/all/src/main/java/org/infinispan/commons/io/ByteBufferFactoryImpl.java

```diff
@@ -8,6 +8,6 @@
 
    @Override
    public ByteBuffer newByteBuffer(byte[] b, int offset, int length) {
-      return new ByteBufferImpl(b, offset, length);
+      return b == null || b.length == 0 || length == 0 ? ByteBufferImpl.EMPTY_INSTANCE : new ByteBufferImpl(b, offset, length);
```

Comment:
Can be:

```java
return ByteBufferImpl.create(b, offset, length);`
```

Reply:
there is no `ByteBufferImpl.create(b, offset, length);` method. Are you suggesting to create one?

Reply:
Sorry, I was thinking there was. But yeah lets add `ByteBufferImpl.create(b, offset, length)`, that way all of the logic around `EMPTY_INSTANCE` is in one place and `ByteBufferImpl.create(b)` impl can just call the overloaded method.

Reply:
done! 
I changed `PersistenceMarshallerImpl` a little since it was allocating and copying the array twice.

Reply:
Good spot, thanks